examples/abitest/module_0/rust/src/lib.rs

```diff
@@ -145,109 +145,180 @@ pub extern "C" fn frontend_oak_main(_in_handle: u64) {
-        tests.insert("ChannelCreateRaw", FrontendNode::test_channel_create_raw);
-        tests.insert("ChannelCreate", FrontendNode::test_channel_create);
-        tests.insert("ChannelCloseRaw", FrontendNode::test_channel_close_raw);
-        tests.insert("ChannelClose", FrontendNode::test_channel_close);
-        tests.insert("ChannelReadRaw", FrontendNode::test_channel_read_raw);
-        tests.insert("ChannelRead", FrontendNode::test_channel_read);
-        tests.insert("ChannelReadOrphan", FrontendNode::test_channel_read_orphan);
-        tests.insert("ChannelWriteRaw", FrontendNode::test_channel_write_raw);
-        tests.insert("ChannelWrite", FrontendNode::test_channel_write);
+        let mut tests: HashMap<&str, (TestFn, Count)> = HashMap::new();
+        tests.insert(
+            "ChannelCreateRaw",
+            (Self::test_channel_create_raw, Count::Unchanged),
+        );
+        tests.insert(
+            "ChannelCreate",
+            (Self::test_channel_create, Count::Unchanged),
+        );
+        tests.insert(
+            "ChannelCloseRaw",
+            (Self::test_channel_close_raw, Count::Unchanged),
+        );
+        tests.insert("ChannelClose", (Self::test_channel_close, Count::Unchanged));
+        tests.insert(
+            "ChannelReadRaw",
+            (Self::test_channel_read_raw, Count::Unchanged),
+        );
+        tests.insert("ChannelRead", (Self::test_channel_read, Count::Unchanged));
+        tests.insert(
+            "ChannelReadOrphan",
+            (Self::test_channel_read_orphan, Count::Unchanged),
+        );
+        tests.insert(
+            "ChannelWriteRaw",
+            (Self::test_channel_write_raw, Count::Unchanged),
+        );
+        tests.insert("ChannelWrite", (Self::test_channel_write, Count::Unchanged));
         tests.insert(
             "ChannelWriteHandle",
-            FrontendNode::test_channel_write_handle,
+            (Self::test_channel_write_handle, Count::Unchanged),
         );
         tests.insert(
             "ChannelWriteOrphanEmpty",
-            FrontendNode::test_channel_write_orphan_empty,
+            (Self::test_channel_write_orphan_empty, Count::Unchanged),
         );
         tests.insert(
             "ChannelWriteOrphanFull",
-            FrontendNode::test_channel_write_orphan_full,
+            (Self::test_channel_write_orphan_full, Count::Unchanged),
         );
         tests.insert(
             "ChannelHandleRecovered",
-            FrontendNode::test_channel_handle_recovered,
+            (Self::test_channel_handle_recovered, Count::Unchanged),
+        );
+        tests.insert(
+            "ChannelChainLost",
+            (Self::test_channel_chain_lost, Count::Unchanged),
+        );
+        tests.insert(
+            "ChannelChainLeaked",
+            (
+                Self::test_channel_chain_leaked,
+                Count::Delta {
+                    nodes: 0,
+                    channels: 9,
```

Comment:
The count values can't be checked in the body of the test because the tests run inside Oak, where the only API functionality available is the various [host functions](https://github.com/project-oak/oak/blob/main/docs/abi.md#host-functions).  The methods to return current object counts are only accessible *outside* Oak, in the code that runs the Runtime (via the `oak_tests` helpers).

Returning the expected counts from each test is a nice idea, but unfortunately it doesn't quite work: we still need a way to check which tests are predictable *in advance* of running the test (line 336). So we'd still need a deterministic flag in the test registry anyway, at which point it seemed better to keep the `deterministic : bool` in the same place as the `(i32, i32)` for expected count changes.